rules/integrations/aws/defense_evasion_vpc_security_group_ingress_rule_added_for_remote_connections.toml

```diff
@@ -0,0 +1,105 @@
+
+### False Positive Analysis:
+
+- **Legitimate Administrative Actions**: Verify if the ingress rule change aligns with scheduled updates, maintenance activities, or legitimate administrative tasks documented in change management tickets or systems.
+- **Consistency Check**: Compare the action against historical data of similar actions performed by the user or within the organization. Consistency with past legitimate actions might indicate a false alarm.
+- **Verify through Outcomes**: Check the `aws.cloudtrail.response_elements` and the `event.outcome` to confirm if the change was successful and intended as per policy.
+
+### Response and Remediation:
+
+- **Immediate Review and Reversal if Necessary**: If the change was unauthorized, revert the security group rules to their previous state to close any unintended access.
+- **Enhance Monitoring and Alerts**: Adjust monitoring systems to alert on similar security group changes, especially those that open access to well-known ports from any IP address.
+- **Educate and Train**: Provide additional training to users with administrative rights on the importance of security best practices concerning security group management.
+- **Audit Security Groups and Policies**: Conduct a comprehensive audit of all security groups and associated policies to ensure they adhere to the principle of least privilege.
+- **Incident Response**: If there's an indication of malicious intent or a security breach, initiate the incident response protocol to mitigate any damage and prevent future occurrences.
+
+### Additional Information:
+
+For further guidance on managing security group rules and securing AWS environments, refer to the [Amazon VPC Security Groups documentation](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html) and AWS best practices for security.
+
+"""
+references = [
+    "https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_AuthorizeSecurityGroupEgress.html",
+    "https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_AuthorizeSecurityGroupIngress.html",
+    "https://www.linkedin.com/pulse/my-backdoors-your-aws-infrastructure-part-3-network-micha%C5%82-brygidyn/",
+]
+risk_score = 47
+rule_id = "25e7fee6-fc25-11ee-ba0f-f661ea17fbce"
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Data Source: AWS",
+    "Data Source: Amazon Web Services",
+    "Data Source: AWS EC2",
+    "Use Case: Threat Detection",
+    "Tactic: Defense Evasion",
+]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset: "aws.cloudtrail" and event.provider: ec2.amazonaws.com
+    and event.action: AuthorizeSecurityGroupIngress and event.outcome: success
+    and (aws.cloudtrail.flattened.request_parameters.cidrIp: 0.0.0.0/0 or aws.cloudtrail.flattened.request_parameters.fromPort: (22 or 3389))
```

Comment:
```suggestion
event.dataset:"aws.cloudtrail" and event.provider:ec2.amazonaws.com
    and event.action:AuthorizeSecurityGroupIngress and event.outcome:success
    and (aws.cloudtrail.flattened.request_parameters.cidrIp:0.0.0.0/0 or aws.cloudtrail.flattened.request_parameters.fromPort:(22 or 3389))
```

Reply:
also do you know if `aws.cloudtrail.flattened.request_parameters.cidrIp:0.0.0.0/0` can be circumvented with `00.0.0.0/0` or similar variations, or does AWS normalize first

Reply:
That is a great question. I would need to test this to confirm

Reply:
Confirmed that AWS normalizes this.